### PR TITLE
Add Toast

### PR DESCRIPTION
**Pull request by joshgachnang**
### **User description**
Skipping buttons. They're extra small buttons with different colors, and we don't use them currently. 
Skipping the second design for toasts as well.

Screenshot 2024-06-28 at 2 39 53 PM

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Updated Toast stories to use persistent toast components and added new examples.
- Extended `ToastProps` interface with new properties such as `title`, `variant`, `secondary`, `size`, `onDismiss`, `persistent`, and `subtitle`.
- Fixed the passing of toast options in `FernsProvider`.
- Simplified `onClick` prop types for buttons in `Modal`.
- Refactored `Toast` component to support new properties, added new variants and sizes, and improved layout and styling.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Toast.stories.tsx</strong><dd><code>Update Toast stories with persistent examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
apps/demo/stories/Toast.stories.tsx

<li>Replaced button-triggered toasts with persistent toast components.<br> <li> Simplified toast variants and added new examples.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/605/files#diff-9550ea2ad6140c2942aa1ff8752eb750971b7e318401c11d03bacb74d4b57d9a">+65/-107</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Common.ts</strong><dd><code>Extend ToastProps interface with new properties</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/src/Common.ts

<li>Updated <code>ToastProps</code> interface to include new properties.<br> <li> Removed <code>message</code> and added <code>title</code>, <code>variant</code>, <code>secondary</code>, <code>size</code>, <code>onDismiss</code>, <br><code>persistent</code>, and <code>subtitle</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/605/files#diff-538e93b553971318d69ff6fff2ac103f295c4922e24bbe61c30897baa463bba1">+11/-8</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Toast.tsx</strong><dd><code>Refactor Toast component with new properties and styling</code>&nbsp; </dd></summary>
<hr>
      
packages/ui/src/Toast.tsx

<li>Refactored <code>Toast</code> component to support new properties.<br> <li> Added new toast variants and sizes.<br> <li> Improved layout and styling.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/605/files#diff-758dac505436c671757c95807bc3373de17286f290521575aae6c1446adee67a">+191/-69</a></td>

</tr>                    
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>FernsProvider.tsx</strong><dd><code>Fix toast options passing in FernsProvider</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/src/FernsProvider.tsx

- Modified `renderToast` to correctly pass toast options.

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/605/files#diff-b49d2acfa459e2bfd0ac98ebbf945daa99e6f678371278fe4df2c0cd9b0138ee">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Modal.tsx</strong><dd><code>Simplify button onClick prop types in Modal</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/src/Modal.tsx

- Simplified onClick prop types for buttons.

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/605/files#diff-3ab16d64d26e5ea343f785756eac0fdd776a49367e2c34dbf6b99bc524a0789f">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: packages/ui/src/Toast.tsx [hunk 1]
```diff
@@ -53,58 +79,154 @@ export function useToast(): {
...
+  const {theme} = useTheme();
+  let color: SurfaceColor;
+  let textColor: TextColor;
+  let iconName: IconName;
+
+  if (secondary) {
+    throw new Error("Secondary not supported yet");
   }
+
+  // if ((buttonText && !buttonOnClick) || (!buttonText && buttonOnClick)) {
```
Comment (hchamorro) [inline packages/ui/src/Toast.tsx:102]: left over?

File: packages/ui/src/Toast.tsx [hunk 2]
```diff
@@ -53,58 +79,154 @@ export function useToast(): {
...
+          accessibilityRole="button"
+          style={{
+            display: "flex",
+            alignItems: "center",
+            alignSelf: "center",
+            gap: 12,
+            marginLeft: 10,
+            padding: size === "lg" ? 8 : 0,
+          }}
+          onPress={onDismiss}
```
Comment (hchamorro) [inline packages/ui/src/Toast.tsx:225]: In demo, pressing did not dismiss the toast sooner

**Review by hchamorro — APPROVED** [review]
Looks good! One thing...
Pressing X in Demo did not dismiss the toast sooner

File: packages/ui/src/Toast.tsx (hunk 2 above)
Comment (joshgachnang) [reply packages/ui/src/Toast.tsx:225]: Ahh, it's because there shouldn't be a dismiss unless it's persistent.

**Pull request merged**